arbos/tx_processor.go

```diff
@@ -197,26 +200,33 @@ func (p *TxProcessor) StartTxHook() (endTxNow bool, gasUsed uint64, err error, r
 		balance := statedb.GetBalance(tx.From)
 		basefee := evm.Context.BaseFee
 		usergas := p.msg.Gas()
-		gascost := arbmath.BigMulByUint(basefee, usergas)
-
-		if arbmath.BigLessThan(balance, gascost) || usergas < params.TxGas {
-			// user didn't have or provide enough gas to do an initial redeem
-			return true, 0, nil, underlyingTx.Hash().Bytes()
-		}
 
-		if arbmath.BigLessThan(tx.GasFeeCap, basefee) && p.msg.RunMode() != types.MessageGasEstimationMode {
-			// user's bid was too low
+		maxGasCost := arbmath.BigMulByUint(tx.GasFeeCap, usergas)
+		maxFeePerGasTooLow := arbmath.BigLessThan(tx.GasFeeCap, basefee) && (p.msg.RunMode() != types.MessageGasEstimationMode || tx.GasFeeCap.BitLen() > 0)
```

Comment:
We probably want a comment here to explain this because there's really 2 ideas happening at once